src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -939,7 +939,7 @@ public static IList<Type> GetTypes(this Type type, bool includeMethods = false)
         internal static bool IsOrleansOrReferencesOrleans(Assembly assembly)
         {
             return DoReferencesContain(assembly.GetReferencedAssemblies(), OrleansCoreAssembly)
-                   || string.Equals(assembly.FullName, OrleansCoreAssembly, StringComparison.Ordinal);
+                   || string.Equals(assembly.GetName().Name, OrleansCoreAssembly.Name, StringComparison.Ordinal);
```

Comment:
I agree with the fix on `DoReferencesContain`, but I'm not sure this one is appropriate too. If for some weird reason there are 2 different versions of Orleans, we don't want to load both, do we?
I think we only need to be loosely coupled to the version if an assembly depends on an older Orleans, but not with the Orleans binary itself.